lib/improver/utilities/cube_manipulation.py

```diff
@@ -157,51 +152,435 @@ def _strip_var_names(cubes):
+                           ' will delete it'
+                           ' - value is {}'.format(cube.attributes[key]))
+                    warnings.warn(msg)
+                    cube.attributes.pop(key)
+    return cubes
+
+
+def _equalise_cube_coords(cubes):
+    """
+    Function to equalise coordinates that do not match.
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to check the coords and revise.
+    Returns:
+        cubelist : Iris cubelist
+            List of cubes with revised coords.
+            The number of cubes in cubelist
+            may be greater than the original number
+            of cubes as the original cubes will be sliced
+            so that that they can be merged together.
+            Merging can only create new coords not add
+            to existing mismatching coords.
+            Note: This internal function modifies the incoming cubes
+    Raises:
+        ValueError: If coordinates in error_keys do not match.
+        ValueError: If model_id has more than one point.
+    """
+    unmatching_coords = compare_coords(cubes)
+    if len(unmatching_coords) == 0:
+        cubelist = cubes
+    else:
+        # Check unmatching not in error_keys.
+        error_keys = ['percentile_over', 'threshold']
+        for error_key in error_keys:
+            for key in ([keyval for cube_dict in unmatching_coords
+                         for keyval in cube_dict]):
+                if error_key in key:
+                    msg = ("{} ".format(error_key) +
+                           "coordinates must match to merge")
+                    raise ValueError(msg)
+
+        cubelist = iris.cube.CubeList([])
+        for i, cube in enumerate(cubes):
+            slice_over_keys = []
+            for key in unmatching_coords[i]:
+                # mismatching model id
+                if key == 'model_id':
+                    realization_found = False
+                    # Check to see if there is a mismatch realization coord.
+                    for j, check in enumerate(unmatching_coords):
+                        if 'realization' in check:
+                            realization_found = True
+                            realization_coord = cubes[j].coord('realization')
+                            break
+                    # If there is add model_realization coord
+                    # and realization coord if necessary.
+                    if realization_found:
+                        if len(cube.coord('model_id').points) != 1:
+                            msg = ("Model_id has more than one point")
+                            raise ValueError(msg)
+                        else:
+                            model_id_val = cube.coord('model_id').points[0]
+                        if cube.coords('realization'):
+                            unmatch = unmatching_coords[i]['realization']
+                            if unmatch['data_dims'] >= 0:
```

Comment:
Is this `if` block still necessary, as won't `compare_coords` return either a positive value or a `None`, so this `if` block is equivalent to:
```python
data_dims = unmatch[`data_dims`]
```